R/defer.R

```diff
@@ -16,22 +13,8 @@ defer <- function(expr, envir = parent.frame()) {
 
 }
 
-deferred_run <- function(envir = parent.frame()) {
```

Comment:
This felt superfluous since we have `renv_defer_execute()` but let me know if you'd like to keep it.

I also found a separate way to allow `done` to run deferred handlers on the global environment; see below in the PR and let me know if that's too gross or not.